src/js/view/view.js

```diff
@@ -604,134 +441,52 @@ define([
             if (bool) {
                 // ignore model that triggered this event and use current state model
                 _stateHandler(_instreamModel || _model);
+                if (_controls) {
+                    var breakPoint = setBreakpoint(_playerElement, _lastWidth, _lastHeight);
+                    _controls.resize(_model, breakPoint);
+                }
             }
 
-            var breakPoint = setBreakpoint(_playerElement, _lastWidth, _lastHeight);
-            _setTimesliderFlags(breakPoint, _model.get('audioMode'), bool);
-
             utils.toggleClass(_playerElement, 'jw-flag-controls-disabled', !bool);
         };
 
-        function _doubleClickFullscreen() {
-            if (_model.get('controls')) {
-                _api.setFullscreen();
-            }
-        }
+        this.addControls = function (controls) {
+            _controls = controls;
 
-        function _setupControls() {
             var overlaysElement = _playerElement.querySelector('.jw-overlays');
             overlaysElement.addEventListener('mousemove', _userActivityCallback);
 
-            _displayClickHandler = new ClickHandler(_model, _videoLayer, { useHover: true });
-            _displayClickHandler.on('click', function () {
-                forward({ type: events.JWPLAYER_DISPLAY_CLICK });
-                if (_model.get('controls')) {
-                    _api.play(reasonInteraction());
-                }
-            });
-            _displayClickHandler.on('tap', function () {
-                forward({ type: events.JWPLAYER_DISPLAY_CLICK });
-                _touchHandler();
+            controls.on('uiActivity', function(/* showing */) {
+                _captionsRenderer.renderCues(true);
             });
-            _displayClickHandler.on('doubleClick', _doubleClickFullscreen);
-            _displayClickHandler.on('move', _userActivityCallback);
-            _displayClickHandler.on('over', _userActivityCallback);
-
-            _controlsLayer.appendChild(createDisplayContainer());
 
-            _dock = new Dock(_model);
+            _logo.setup(controls.right);
 
-            _logo = new Logo(_model);
-            _logo.on(events.JWPLAYER_LOGO_CLICK, _logoClickHandler);
+            controls.enable(_api, _model, _videoLayer);
+            
+            _componentFadeListeners(controls.controlbar);
```

Comment:
`_componentFadeListeners` should be moved to controls. These listeners need to be added and removed by controls.